qiskit/transpiler/passes/optimization/optimize_1q_commutation.py

```diff
@@ -227,7 +223,8 @@ def _step(self, dag):
                     + (new_run or QuantumCircuit(1)).data
                     + (new_succeeding_run or QuantumCircuit(1)).data
                 ),
-                new_basis + new_preceding_basis + new_succeeding_basis,
+                self._optimize1q._basis_gates,
+                dag.qubits.index(run[0].qargs[0]),
```

Comment:
This will get expensive over time because each loop iteration it's an `O(n)` lookup for the index. Typically the pattern we've used to avoid this to just build a mapping outside the for loop like:

```
qubit_indices = {bit: index for index, bit in enumerate(dag.qubits)
```

and then just change this line to be: `qubit = qubit_indices[run[0].qargs[0]]`

Reply:
fixed